cookbooks/bcpc/recipes/ssh.rb

```diff
@@ -74,118 +60,82 @@
   }
 end
 
-ssh_key_types = %w{dsa ecdsa ed25519 rsa}
-
-# First, try to retrieve existing host keys from the server.
-ruby_block 'get-ssh-secrets-from-server' do
-  block do
-    require 'chef-vault'
-    begin
-      node.run_state[:bcpc_ssh_host_keys] =
-        ChefVault::Item.load('ssh_host_keys', node[:fqdn])
-    rescue Exception => ee
-      # If we fail to load the vault, log the failure and create a vault.
-      Chef::Log.warn("Failed to load ssh_host_keys/#{node[:fqdn]}:\n#{ee}")
-
-      begin
-        vault_item = ChefVault::Item.new('ssh_host_keys', node[:fqdn])
-        vault_item.admins([get_bootstrap, node[:fqdn], 'admin'].join(','))
-        vault_item.search("fqdn:#{node[:fqdn]}")
-        vault_item.save
-        node.run_state[:bcpc_ssh_host_keys] = vault_item
-      rescue Exception => eee
-        # If we also fail to create, this node must not be an admin.
-        Chef::Log.warn('Failed to create new vault ' \
-                       "ssh_host_keys/#{node[:fqdn]}. " \
-                       "Is this node an admin?\n#{eee}")
-      end
-    end
-  end
-  not_if { node[:fqdn] == get_bootstrap }
+require 'chef-vault'
+
+def load_ssh_keys
+  ChefVault::Item.load('ssh_host_keys', node[:fqdn])
+rescue Exception => ee
+  Chef::Log.warn("Failed to load ssh_host_keys/#{node[:fqdn]}:\n#{ee}")
+  return nil
 end
 
-#
-# If we failed to get host keys, read the files on disk and write them
-# back to the chef server.
-#
-# (This block will also update the server with new key types.)
-#
-ruby_block 'read-ssh-secrets' do
-  block do
-    vault_item = node.run_state[:bcpc_ssh_host_keys]
-
-    if vault_item.is_a?(ChefVault::Item)
-      ssh_key_types.each do |key_type|
-        private_key_path = "/etc/ssh/ssh_host_#{key_type}_key"
-        public_key_path = "/etc/ssh/ssh_host_#{key_type}_key.pub"
-
-        if vault_item[key_type].nil? && ::File.exist?(private_key_path)
-          Chef::Log.info("Saving #{key_type} for #{node[:fqdn]}")
-          vault_item[key_type] = ::File.read(private_key_path)
-          vault_item.save
-        end
-      end
-    else
-      Chef::Log.warn("No vault item found for ssh_host_keys/#{node[:fqdn]}!")
-    end
-  end
-  not_if { node[:fqdn] == get_bootstrap }
+def create_ssh_keys
+  vault_item = ChefVault::Item.new('ssh_host_keys', node[:fqdn])
+  vault_item.admins([get_bootstrap, node[:fqdn], 'admin'].join(','))
+  vault_item.search("fqdn:#{node[:fqdn]}")
+  vault_item.save
+rescue Exception => eee
```

Comment:
I do not think `eee` is common in our code for exceptions. Don't we use `e` or `ex` normally?